### ISSUE TITLE
[BUG] Bun doesnt Ignore SSL issues with rejectUnauthorized

### ISSUE THREAD

**Issue by kravetsone** [oven-sh/bun#11821]
### What version of Bun is running?

1.1.12+43f0913c3

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

When we develop in Dev mode, we are needed to use a VPN

```js
import https from "node:https";

// two ways of disabling UNABLE_TO_GET_ISSUER_CERT_LOCALLY errors
process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
https.globalAgent.options.rejectUnauthorized = false;

const response = await fetch(
	process.env.SOME_LINK
);

const data = await response.text();

console.log(data.length);
```

Run with [Bun](https://bun.sh/)

```ts
PS Z:\PROJECTS\node-ts\elycart-monorepo> bun .\1.mjs
UNABLE_TO_GET_ISSUER_CERT_LOCALLY: unable to get local issuer certificate
 path: "https://130.193.66.56/RuRu.FrontEnd.ServiceProvider2/TransactionService.svc"

Bun v1.1.13 (Windows x64)
```

Run with Node.js

```ts
PS Z:\PROJECTS\node-ts\elycart-monorepo> node 1.mjs  
(node:33948) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
35183
```

With https.Agent too:

```ts
const httpsAgent = new https.Agent({
  rejectUnauthorized: false,
});

axios.get(process.env.SOME_LINK, { httpsAgent });
```

**Comment by paperdave** [comment]
i thought NODE_TLS_REJECT_UNAUTHORIZED=0 works, but it might have to be passed as an environment variable from outside, not being the mutated global. workaround should be:

```
NODE_TLS_REJECT_UNAUTHORIZED=0 bun run ./index.ts
```

since the two options shown do work in node, we should make sure this works in bun too

**Comment by kravetsone** [comment]
> NODE_TLS_REJECT_UNAUTHORIZED=0

Ohh)
Thank you! it works for me

**Comment by Jarred-Sumner** [comment]
This is fixed in #11833
